server/utils/resident/placement.test.ts

```diff
@@ -0,0 +1,182 @@
+import { characteristicsBulletList } from '../characteristicsUtils'
+import { placementStatusTag } from '../placements'
+
+describe('tabController', () => {
+  const allPlacementStatuses = [
+    [cas1SpaceBookingFactory.upcoming().build()],
+    [cas1SpaceBookingFactory.current().build()],
+    [cas1SpaceBookingFactory.nonArrival().build()],
+    [cas1SpaceBookingFactory.departed().build()],
+    [cas1SpaceBookingFactory.departedBreachOrRecall().build()],
+    [cas1SpaceBookingFactory.departedPlannedMoveOn().build()],
+    [cas1SpaceBookingFactory.cancelled().build()],
+  ]
+
+  it.each(allPlacementStatuses)(
+    'should return a details card contain the premises name, date allocated, status, and delius number for all placements',
+    placement => {
+      expect(placementTabController(placement)).toEqual({
+        subHeading: `${placement.premises.name} AP placement`,
+        cardList: [
+          {
+            card: { title: { text: placement.premises.name } },
+            rows: expect.arrayContaining([
+              {
+                key: { text: 'Approved Premises' },
+                value: { text: placement.premises.name },
+              },
+              {
+                key: { text: 'Date allocated' },
+                value: { text: DateFormats.isoDateToUIDate(placement.createdAt, { format: 'short' }) },
+              },
+              {
+                key: { text: 'Status' },
+                value: { html: placementStatusTag(placement) },
+              },
+              {
+                key: { text: 'Delius event number' },
+                value: { text: placement.deliusEventNumber },
+              },
+            ]),
+          },
+        ],
+      })
+    },
+  )
+
+  it.each(allPlacementStatuses)(
+    'should return a details card contain the AP and room requirements for all placements',
+    placement => {
+      expect(placementTabController(placement)).toEqual({
+        subHeading: `${placement.premises.name} AP placement`,
+        cardList: [
+          {
+            card: { title: { text: placement.premises.name } },
+            rows: expect.arrayContaining([
+              {
+                key: { text: 'AP requirements' },
+                value: { html: characteristicsBulletList(filterApLevelCriteria(placement.characteristics)) },
+              },
+              {
+                key: { text: 'Room requirements' },
+                value: { html: characteristicsBulletList(filterRoomLevelCriteria(placement.characteristics)) },
+              },
+            ]),
+          },
+        ],
+      })
+    },
+  )
+
+  it.each([
+    [cas1SpaceBookingFactory.upcoming().build(), true],
+    [cas1SpaceBookingFactory.nonArrival().build(), true],
+    [cas1SpaceBookingFactory.current().build(), false],
+    [cas1SpaceBookingFactory.departed().build(), false],
+  ])('should show expected arrival date for placements that have not arrived', (placement, shouldContain) => {
```

Comment:
It's nice to make the first parameter a string and add it to the description, so that you can see the tests that are run more easily.